nebula-logger/tests/log-management/classes/LogEntryEventHandler_Tests.cls

```diff
@@ -474,18 +215,58 @@ private class LogEntryEventHandler_Tests {
         }
     }
 
+    // TODO - testing topics is tricky. Within the unlocked package, we can test it,
+    // but for people that prefer using the unpackaged metadata + don't use topics,
+    // the tests would fail. Need to further investigate ways to handle this
+    // @isTest
+    // static void it_should_create_topic_and_topicAssignment_records_when_enabled() {
+    //     String transactionId = '123-456-789-0';
+
+    //     List<String> tags = new List<String>{ 'test-tag-1', 'test-tag-2' };
+    //     LogEntryEvent__e logEntryEvent = new LogEntryEvent__e(
+    //         EpochTimestamp__c = System.now().getTime(),
+    //         Message__c = 'my message',
+    //         Timestamp__c = System.now(),
+    //         Tags__c = String.join(tags, '\n'),
+    //         TransactionEntryNumber__c = 1,
+    //         TransactionId__c = transactionId
+    //     );
+
+    //     Test.startTest();
+    //     LogEntryEventHandler.TAGGING_IS_ENABLED = true;
+    //     LogEntryEventHandler.TAG_USING_TOPICS = true;
+    //     Database.SaveResult saveResult = EventBus.publish(logEntryEvent);
+    //     Test.stopTest();
+
+    //     System.assertEquals(true, saveResult.isSuccess(), saveResult.getErrors());
+
+    //     Log__c log = getLog();
+    //     System.assertEquals(1, log.LogEntries__r.size());
+    //     LogEntry__c logEntry = log.LogEntries__r.get(0);
+
+    //     System.assertEquals(logEntryEvent.TransactionId__c, log.TransactionId__c);
+    //     validateLogFields(logEntryEvent, log);
+    //     validateLogEntryFields(logEntryEvent, logEntry);
+
+    //     Set<String> tagsSet = new Set<String>(tags);
+    //     List<TopicAssignment> topicAssignments = [SELECT Id, EntityId, TopicId, Topic.Name FROM TopicAssignment WHERE EntityId = :logEntry.Id];
+    //     for (TopicAssignment topicAssignment : topicAssignments) {
+    //         System.assert(tagsSet.contains(topicAssignment.Topic.Name));
+    //     }
+    // }
+
     @isTest
-    static void it_should_create_topic_and_topicAssignment_records() {
-        String transactionId = '123-456-789-0';
+    static void it_should_create_tag_records_when_enabled() {
+        System.assertEquals(true, LogEntryEventHandler.TAGGING_IS_ENABLED);
```

Comment:
I don't always have custom assert messages, but especially since this is at the start of the test I think something about the test having started under the wrong conditions helps to clarify what's happening in the event of a premature failure

Reply:
Yeah, that's completely fair - I need to be better about adding assert messages. I don't think they're needed for every assert, but there are definitely some asserts in this repo that would benefit from a custom message.

Reply:
Custom assert messages have been added! 🎆